src/push.ts

```diff
@@ -0,0 +1,149 @@
+import { defer, Observable, Subject } from 'rxjs';
+import { Client } from '.';
+import { ENTITY_TYPES, IEntityType, IStateChange, IWebSocketPushEnable } from './types';
+import { Transport } from './utils/transport';
+import * as ws from 'isomorphic-ws';
+
+export class PushClient {
+  private client: Client;
+  private transport: Transport;
+  private httpHeaders: { [headerName: string]: string };
+  private webSocket?: ws;
+  private webSocketSubjects: { [_ in IEntityType]: Subject<{ [accountId: string]: string }> };
+
+  private started?: Promise<void>;
+  private enabledDataTypes: { [_ in IEntityType]?: boolean };
+
+  constructor({
+    client,
+    transport,
+    httpHeaders,
+  }: {
+    client: Client;
+    transport: Transport;
+    httpHeaders: { [headerName: string]: string };
+  }) {
+    this.client = client;
+    this.transport = transport;
+    this.httpHeaders = httpHeaders;
+    this.webSocketSubjects = {
+      Mailbox: new Subject(),
+      Email: new Subject(),
+      EmailSubmission: new Subject(),
+    };
+    this.enabledDataTypes = {};
+  }
+
+  public start(): Promise<void> {
+    if (!this.started) {
+      this.started = new Promise(resolve => {
+        this.transport
+          .post<{
+            value: string;
+          }>(`${this.client.getApiUrl()}/ws/ticket`, '', this.httpHeaders)
+          .then(response => {
+            const ticket = response.value;
```

Comment:
Im not sure this non standard auth sheme should ve hard coded as the default.

Reply:
There is currently no alternative as far as I know.
I would be glad to add another way to open a websocket.

I also replaced the hard-coded url by the capability one.